internals/overlord/servstate/manager.go

```diff
@@ -264,7 +264,15 @@ func (m *ServiceManager) Replan() ([][]string, [][]string, error) {
 			if config.Equal(s.config) {
 				continue
 			}
-			s.config = config.Copy() // update service config from plan
+			// Update service config and workload from plan
+			s.config = config.Copy()
+			if s.config.Workload != "" {
```

Comment:
If workload changed from "foo" to "" with the previous update, I think you do want to update ```s.workload``` to not contain a copy of the previous workload ?

Reply:
Also if the plan.Service is the same, and the workload changed, replan will not restart the service.

Reply:
Nice catch!